src/rdiff_backup/connection.py

```diff
@@ -157,10 +159,11 @@ class LowLevelPipeConnection(Connection):
 
     """
 
-    def __init__(self, inpipe, outpipe):
+    def __init__(self, inpipe, outpipe, process=None):
```

Comment:
Checking again `testing/connectiontest.py`, I think that `self.process` should stay at `PipeConnection` level and shouldn't be known by LowLevelPipeConnection. Basically, this is the difference between both that one is only two input and output pipes, whereas the other really has a process in-between.

That means that a _close method of PipeConnection should override the LowLevel one and contain your code. Does it make sense?